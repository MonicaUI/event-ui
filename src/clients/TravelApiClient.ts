import { RequestBuilder } from '@cvent/nucleus-networking';
import { SESSION_HEADER_EVENT_ID, SESSION_HEADER_USER_TYPE } from '@cvent/event-ui-networking';
import Logger from '@cvent/nucleus-logging';
import { fetchWithSessionTimeout } from '../dialogs/SessionTimedOutDialog';
import { ServiceError } from '@cvent/event-ui-networking';
import { GET_HOTEL_ROOM_CAPACITIES } from 'event-widgets/lib/HotelRequest/useHotelRoomCapacity';
import { ApolloClient, InMemoryCache } from '@apollo/client';
import { HotelRoomCapacities } from 'event-widgets/redux/modules/eventTravel';

const LOG = new Logger('TravelApiClient');

/**
 * The travel cart service client.
 */
export default class TravelApiClient {
  travelApiBaseUrl;
  travelCartBaseUrl;
  environment;
  request;
  createdBy;
  constructor(baseURL: $TSFixMe, eventId: $TSFixMe, environment: $TSFixMe, userType: $TSFixMe, createdBy: $TSFixMe) {
    this.travelApiBaseUrl = baseURL + 'travel-booking/v1';
    this.travelCartBaseUrl = this.travelApiBaseUrl + '/travel-carts';
    this.environment = environment;
    this.createdBy = createdBy;
    this.request = new RequestBuilder({ url: this.travelApiBaseUrl }).withCookies();
    this.request = this.request.header(SESSION_HEADER_EVENT_ID, eventId).header(SESSION_HEADER_USER_TYPE, userType);
  }

  /**
   * get roommates
   * @param eventId
   * @param params search parameters for post body
   * @returns {Promise<*>}
   */
  async getRoommates(eventId: $TSFixMe, params: $TSFixMe, travelSnapshotVersion: $TSFixMe): Promise<$TSFixMe> {
    const request = this.request
      .url(`${this.travelApiBaseUrl}/roommates/${eventId}`)
      .header('Travel-Snapshot-Version', travelSnapshotVersion)
      .query('environment', this.environment)
      .post()
      .json(params)
      .build();
    const response = await fetchWithSessionTimeout(request);
    if (!response.ok) {
      const error = await ServiceError.create('Error fetching roommates', response, request);
      LOG.info('Error fetching roommates', request.url, error.responseStatus, error.responseBody);
      throw error;
    }
    return await response.json();
  }

  /**
   * Creates a travel cart
   * @param travelCart the travel cart to create
   */
  async createTravelCart(travelCart: $TSFixMe): Promise<$TSFixMe> {
    const request = this.request
      .url(this.travelCartBaseUrl)
      .header('Travel-Snapshot-Version', travelCart.travelSnapshotVersion)
      .header('Event-Snapshot-Version', travelCart.eventSnapshotVersion)
      .header('Created-By', this.createdBy)
      .query('environment', this.environment)
      .post()
      .json(travelCart)
      .build();
    const response = await fetchWithSessionTimeout(request);
    if (!response.ok) {
      const error = await ServiceError.create('Error creating travel cart', response, request);
      LOG.info('Error creating travel cart', request.url, error.responseStatus, error.responseBody);
      throw error;
    }
    return await response.json();
  }

  /**
   * Updates a travel cart
   * @param travelCart what to update the travel cart to
   */
  async updateTravelCart(travelCart: $TSFixMe): Promise<$TSFixMe> {
    const requestToBuild = this.request
      .url(`${this.travelCartBaseUrl}/${travelCart.id}`)
      .query('environment', this.environment)
      .put()
      .json(travelCart);
    const request = requestToBuild.build();
    const response = await fetchWithSessionTimeout(request);
    if (!response.ok) {
      const error = await ServiceError.create('Error updating travel cart', response, request);
      LOG.info('Error updating travel cart', request.url, error.responseStatus, error.responseBody);
      throw error;
    }
    return await response.json();
  }

  /**
   * Get a travel cart
   * @param {*} travelCartId the travel cart id corresponding to which the travel cart is to be fetched
   */
  async getTravelCart(travelCartId: $TSFixMe): Promise<$TSFixMe> {
    const request = this.request
      .url(`${this.travelCartBaseUrl}/${travelCartId}`)
      .query('environment', this.environment)
      .get()
      .build();
    const response = await fetchWithSessionTimeout(request);
    if (!response.ok) {
      const error = await ServiceError.create('Error getting travel cart', response, request);
      LOG.info('Error getting travel cart ', request.url, error.responseStatus, error.responseBody);
      throw error;
    }
    return await response.json();
  }

  /**
   * Gets transient travel cart
   * @param inviteeId to get id corresponding to which the transien travel cart needs to be populated
   * @param includeGroupMembers if include group members too in travelCart
   * @returns {Promise<*>}
   */
  async getTransientTravelCart(
    inviteeId: $TSFixMe,
    includeGroupMembers: $TSFixMe,
    travelCartActionRequest: $TSFixMe
  ): Promise<$TSFixMe> {
    const request = this.request
      .url(`${this.travelCartBaseUrl}/transient/${inviteeId}`)
      .query('environment', this.environment)
      .query('includeGroupMembers', includeGroupMembers)
      .post()
      .json(travelCartActionRequest)
      .build();
    const response = await fetchWithSessionTimeout(request);
    if (!response.ok) {
      const error = await ServiceError.create('Error getting travel cart', response, request);
      LOG.info('Error getting travel cart ', request.url, error.responseStatus, error.responseBody);
      throw error;
    }
    return await response.json();
  }

  /**
   * Fetches hotel room capacity summaries
   */
  async getHotelRoomCapacitySummaries(
    isInTestMode: boolean,
    contactTypeId: string,
    apolloClient: ApolloClient<InMemoryCache>
  ): Promise<HotelRoomCapacities> {
    const input = {
      isForTestMode: isInTestMode,
      contactTypeIds: [contactTypeId]
    };
    const result = await apolloClient.query({
      query: GET_HOTEL_ROOM_CAPACITIES,
      variables: {
        input,
        pathFunction: ({ args }) =>
          `${this.travelApiBaseUrl}/capacity/hotel-room-capacity/${args.eventId}?environment=${args.environment}`
      }
    });
    return result?.data?.response;
  }
}
