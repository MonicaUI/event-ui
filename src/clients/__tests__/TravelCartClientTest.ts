import TravelApiClient from '../TravelApiClient';
import travelCartResponse from './fixtures/travelCartResponse.json';
import hotelRoomCapacityResponse from './fixtures/hotelRoomCapacityResponse.json';

jest.mock('@cvent/nucleus-networking', () => {
  const RequestBuilderModule = jest.requireActual<$TSFixMe>('@cvent/nucleus-networking');
  class MockRequestBuilder extends RequestBuilderModule.RequestBuilder {
    constructor(requestOptions = {}) {
      const newRequestOptions = {
        ...requestOptions,
        headers: {
          ...(requestOptions as $TSFixMe).headers,
          httplogpageloadid: 'httplogpageloadid',
          httplogrequestid: 'httplogrequestid'
        }
      };
      super(newRequestOptions);
    }
  }
  let mockResponse = {};
  let mockHeaders = {};
  return {
    ...RequestBuilderModule,
    RequestBuilder: MockRequestBuilder,
    setMockResponse(response, headers) {
      mockResponse = response;
      mockHeaders = headers;
    },
    fetchAndRetryIfServerBusy: jest.fn(() => {
      return Promise.resolve({
        ok: true,
        headers: {
          get(header) {
            return mockHeaders[header];
          },
          has(header) {
            return !!mockHeaders[header];
          }
        },
        clone() {
          return this;
        },
        json: () => Promise.resolve(mockResponse)
      });
    })
  };
});
jest.mock('../../dialogs/SessionTimedOutDialog', () => {
  return {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    fetchWithSessionTimeout: require('@cvent/nucleus-networking').fetchAndRetryIfServerBusy
  };
});

// eslint-disable-next-line @typescript-eslint/no-var-requires
const { setMockResponse } = require('@cvent/nucleus-networking');
const baseUrl = 'https://example.com/travelBase/';
const client = new TravelApiClient(baseUrl, 'eventId', 'dev', 'standard', 'planner');

beforeEach(() => {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  require('@cvent/nucleus-networking').fetchAndRetryIfServerBusy.mockClear();
  let time = 1234;
  // eslint-disable-next-line no-extend-native
  Date.prototype.getTime = () => {
    // eslint-disable-line no-extend-native
    return (time += 1000);
  };
});

test('travelApiClient.createTravelCart matches snapshot', async () => {
  setMockResponse(
    {
      ...travelCartResponse
    },
    {}
  );
  const response = await client.createTravelCart({
    ...travelCartResponse.travelCart
  });

  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const fetch = require('@cvent/nucleus-networking').fetchAndRetryIfServerBusy;
  expect(fetch).toHaveBeenCalled();
  expect(fetch.mock.calls[0]).toMatchSnapshot();
  expect(response).toMatchSnapshot();
});

test('travelApiClient.updateTravelCart matches snapshot', async () => {
  setMockResponse(
    {
      ...travelCartResponse
    },
    {}
  );
  const response = await client.updateTravelCart({
    ...travelCartResponse.travelCart
  });

  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const fetch = require('@cvent/nucleus-networking').fetchAndRetryIfServerBusy;
  expect(fetch).toHaveBeenCalled();
  expect(fetch.mock.calls[0]).toMatchSnapshot();
  expect(response).toMatchSnapshot();
});

test('travelApiClient.getTransientTravelCart matches snapshot', async () => {
  setMockResponse(
    {
      ...travelCartResponse
    },
    {}
  );
  const response = await client.getTransientTravelCart('inviteeId', true, { blah: 'blah' });

  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const fetch = require('@cvent/nucleus-networking').fetchAndRetryIfServerBusy;
  expect(fetch).toHaveBeenCalled();
  expect(fetch.mock.calls[0]).toMatchSnapshot();
  expect(response).toMatchSnapshot();
});

test('travelApiClient.getTravelCart matches snapshot', async () => {
  setMockResponse(
    {
      ...travelCartResponse
    },
    {}
  );
  const response = await client.getTravelCart('inviteeId');

  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const fetch = require('@cvent/nucleus-networking').fetchAndRetryIfServerBusy;
  expect(fetch).toHaveBeenCalled();
  expect(fetch.mock.calls[0]).toMatchSnapshot();
  expect(response).toMatchSnapshot();
});

test('travelApiClient.getRoommates matches snapshot', async () => {
  setMockResponse(
    {
      ...travelCartResponse
    },
    {}
  );
  const response = await client.getRoommates('inviteeId', 'params', 'version-1');

  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const fetch = require('@cvent/nucleus-networking').fetchAndRetryIfServerBusy;
  expect(fetch).toHaveBeenCalled();
  expect(fetch.mock.calls[0]).toMatchSnapshot();
  expect(response).toMatchSnapshot();
});

test('travelApiClient.getHotelRoomCapacitySummaries call with apollo client', async () => {
  const mockHotelRoomCapacityResponse = {
    ...hotelRoomCapacityResponse
  };
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const apolloClient = require('@apollo/client');
  jest.mock('@apollo/client', () => ({
    query: jest.fn(async () => {
      return {
        data: {
          response: mockHotelRoomCapacityResponse
        }
      };
    })
  }));
  const response = await client.getHotelRoomCapacitySummaries(false, 'regTypeId1', apolloClient);
  expect(apolloClient.query).toHaveBeenCalled();
  const args = {
    eventId: 'eventId',
    environment: 'env'
  };
  expect(apolloClient.query.mock.calls[0][0].variables.pathFunction({ args })).toContain(baseUrl);
  expect(response).toEqual(mockHotelRoomCapacityResponse);
});
