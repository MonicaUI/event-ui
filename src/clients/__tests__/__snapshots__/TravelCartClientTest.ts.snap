// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`travelApiClient.createTravelCart matches snapshot 1`] = `
Array [
  Request {
    "_bodyInit": "{\\"id\\":\\"6b0b6913-b477-4a8c-9e7e-481ce5e4b33a\\",\\"status\\":\\"INPROGRESS\\",\\"bookings\\":[{\\"id\\":\\"00000000-0000-0000-0000-000000000000\\",\\"attendee\\":{\\"id\\":\\"614c34fe-f5f6-49d9-8d6c-5c7bc76ee466\\"},\\"registrationTypeId\\":\\"00000000-0000-0000-0000-000000000000\\",\\"hotelRoomBookings\\":[{\\"id\\":\\"00000000-0000-0000-0000-000000000000\\",\\"hotelId\\":\\"b64c34fe-f5f6-49d9-8d6c-5c7bc76ee4b5\\",\\"roomTypeId\\":\\"c26c34fe-f5f6-49d9-8d6c-5c7bc76ee464\\",\\"checkinDate\\":\\"2015-06-22\\",\\"checkoutDate\\":\\"2015-06-23\\",\\"preferences\\":{\\"specialRequestNotes\\":\\"something\\",\\"isSmokingOpted\\":true,\\"rewardsCode\\":\\"something\\",\\"isRoomSharingOpted\\":true,\\"isAdaAccessibilityOpted\\":false,\\"roommateDescription\\":\\"\\"},\\"travellerInfo\\":{\\"gender\\":\\"0\\",\\"firstName\\":\\"\\",\\"lastName\\":\\"\\"},\\"requestedAction\\":\\"BOOK\\"}]}],\\"travelSnapshotVersion\\":\\"abc\\",\\"eventSnapshotVersion\\":\\"def\\",\\"isForPreview\\":false,\\"isForPlanner\\":false}",
    "_bodyText": "{\\"id\\":\\"6b0b6913-b477-4a8c-9e7e-481ce5e4b33a\\",\\"status\\":\\"INPROGRESS\\",\\"bookings\\":[{\\"id\\":\\"00000000-0000-0000-0000-000000000000\\",\\"attendee\\":{\\"id\\":\\"614c34fe-f5f6-49d9-8d6c-5c7bc76ee466\\"},\\"registrationTypeId\\":\\"00000000-0000-0000-0000-000000000000\\",\\"hotelRoomBookings\\":[{\\"id\\":\\"00000000-0000-0000-0000-000000000000\\",\\"hotelId\\":\\"b64c34fe-f5f6-49d9-8d6c-5c7bc76ee4b5\\",\\"roomTypeId\\":\\"c26c34fe-f5f6-49d9-8d6c-5c7bc76ee464\\",\\"checkinDate\\":\\"2015-06-22\\",\\"checkoutDate\\":\\"2015-06-23\\",\\"preferences\\":{\\"specialRequestNotes\\":\\"something\\",\\"isSmokingOpted\\":true,\\"rewardsCode\\":\\"something\\",\\"isRoomSharingOpted\\":true,\\"isAdaAccessibilityOpted\\":false,\\"roommateDescription\\":\\"\\"},\\"travellerInfo\\":{\\"gender\\":\\"0\\",\\"firstName\\":\\"\\",\\"lastName\\":\\"\\"},\\"requestedAction\\":\\"BOOK\\"}]}],\\"travelSnapshotVersion\\":\\"abc\\",\\"eventSnapshotVersion\\":\\"def\\",\\"isForPreview\\":false,\\"isForPlanner\\":false}",
    "credentials": "include",
    "headers": Headers {
      "map": Object {
        "content-type": "application/json",
        "created-by": "planner",
        "cvent-event-id": "eventId",
        "cvent-user-type": "standard",
        "event-snapshot-version": "def",
        "httplogpageloadid": "httplogpageloadid",
        "httplogrequestid": "httplogrequestid",
        "travel-snapshot-version": "abc",
      },
    },
    "method": "POST",
    "mode": null,
    "referrer": null,
    "url": "https://example.com/travelBase/travel-booking/v1/travel-carts?environment=dev",
  },
]
`;

exports[`travelApiClient.createTravelCart matches snapshot 2`] = `
Object {
  "travelCart": Object {
    "bookings": Array [
      Object {
        "attendee": Object {
          "id": "614c34fe-f5f6-49d9-8d6c-5c7bc76ee466",
        },
        "hotelRoomBookings": Array [
          Object {
            "checkinDate": "2015-06-22",
            "checkoutDate": "2015-06-23",
            "hotelId": "b64c34fe-f5f6-49d9-8d6c-5c7bc76ee4b5",
            "id": "00000000-0000-0000-0000-000000000000",
            "preferences": Object {
              "isAdaAccessibilityOpted": false,
              "isRoomSharingOpted": true,
              "isSmokingOpted": true,
              "rewardsCode": "something",
              "roommateDescription": "",
              "specialRequestNotes": "something",
            },
            "requestedAction": "BOOK",
            "roomTypeId": "c26c34fe-f5f6-49d9-8d6c-5c7bc76ee464",
            "travellerInfo": Object {
              "firstName": "",
              "gender": "0",
              "lastName": "",
            },
          },
        ],
        "id": "00000000-0000-0000-0000-000000000000",
        "registrationTypeId": "00000000-0000-0000-0000-000000000000",
      },
    ],
    "eventSnapshotVersion": "def",
    "id": "6b0b6913-b477-4a8c-9e7e-481ce5e4b33a",
    "isForPlanner": false,
    "isForPreview": false,
    "status": "INPROGRESS",
    "travelSnapshotVersion": "abc",
  },
}
`;

exports[`travelApiClient.getRoommates matches snapshot 1`] = `
Array [
  Request {
    "_bodyInit": "\\"params\\"",
    "_bodyText": "\\"params\\"",
    "credentials": "include",
    "headers": Headers {
      "map": Object {
        "content-type": "application/json",
        "cvent-event-id": "eventId",
        "cvent-user-type": "standard",
        "httplogpageloadid": "httplogpageloadid",
        "httplogrequestid": "httplogrequestid",
        "travel-snapshot-version": "version-1",
      },
    },
    "method": "POST",
    "mode": null,
    "referrer": null,
    "url": "https://example.com/travelBase/travel-booking/v1/roommates/inviteeId?environment=dev",
  },
]
`;

exports[`travelApiClient.getRoommates matches snapshot 2`] = `
Object {
  "travelCart": Object {
    "bookings": Array [
      Object {
        "attendee": Object {
          "id": "614c34fe-f5f6-49d9-8d6c-5c7bc76ee466",
        },
        "hotelRoomBookings": Array [
          Object {
            "checkinDate": "2015-06-22",
            "checkoutDate": "2015-06-23",
            "hotelId": "b64c34fe-f5f6-49d9-8d6c-5c7bc76ee4b5",
            "id": "00000000-0000-0000-0000-000000000000",
            "preferences": Object {
              "isAdaAccessibilityOpted": false,
              "isRoomSharingOpted": true,
              "isSmokingOpted": true,
              "rewardsCode": "something",
              "roommateDescription": "",
              "specialRequestNotes": "something",
            },
            "requestedAction": "BOOK",
            "roomTypeId": "c26c34fe-f5f6-49d9-8d6c-5c7bc76ee464",
            "travellerInfo": Object {
              "firstName": "",
              "gender": "0",
              "lastName": "",
            },
          },
        ],
        "id": "00000000-0000-0000-0000-000000000000",
        "registrationTypeId": "00000000-0000-0000-0000-000000000000",
      },
    ],
    "eventSnapshotVersion": "def",
    "id": "6b0b6913-b477-4a8c-9e7e-481ce5e4b33a",
    "isForPlanner": false,
    "isForPreview": false,
    "status": "INPROGRESS",
    "travelSnapshotVersion": "abc",
  },
}
`;

exports[`travelApiClient.getTransientTravelCart matches snapshot 1`] = `
Array [
  Request {
    "_bodyInit": "{\\"blah\\":\\"blah\\"}",
    "_bodyText": "{\\"blah\\":\\"blah\\"}",
    "credentials": "include",
    "headers": Headers {
      "map": Object {
        "content-type": "application/json",
        "cvent-event-id": "eventId",
        "cvent-user-type": "standard",
        "httplogpageloadid": "httplogpageloadid",
        "httplogrequestid": "httplogrequestid",
      },
    },
    "method": "POST",
    "mode": null,
    "referrer": null,
    "url": "https://example.com/travelBase/travel-booking/v1/travel-carts/transient/inviteeId?environment=dev&includeGroupMembers=true",
  },
]
`;

exports[`travelApiClient.getTransientTravelCart matches snapshot 2`] = `
Object {
  "travelCart": Object {
    "bookings": Array [
      Object {
        "attendee": Object {
          "id": "614c34fe-f5f6-49d9-8d6c-5c7bc76ee466",
        },
        "hotelRoomBookings": Array [
          Object {
            "checkinDate": "2015-06-22",
            "checkoutDate": "2015-06-23",
            "hotelId": "b64c34fe-f5f6-49d9-8d6c-5c7bc76ee4b5",
            "id": "00000000-0000-0000-0000-000000000000",
            "preferences": Object {
              "isAdaAccessibilityOpted": false,
              "isRoomSharingOpted": true,
              "isSmokingOpted": true,
              "rewardsCode": "something",
              "roommateDescription": "",
              "specialRequestNotes": "something",
            },
            "requestedAction": "BOOK",
            "roomTypeId": "c26c34fe-f5f6-49d9-8d6c-5c7bc76ee464",
            "travellerInfo": Object {
              "firstName": "",
              "gender": "0",
              "lastName": "",
            },
          },
        ],
        "id": "00000000-0000-0000-0000-000000000000",
        "registrationTypeId": "00000000-0000-0000-0000-000000000000",
      },
    ],
    "eventSnapshotVersion": "def",
    "id": "6b0b6913-b477-4a8c-9e7e-481ce5e4b33a",
    "isForPlanner": false,
    "isForPreview": false,
    "status": "INPROGRESS",
    "travelSnapshotVersion": "abc",
  },
}
`;

exports[`travelApiClient.getTravelCart matches snapshot 1`] = `
Array [
  Request {
    "_bodyInit": undefined,
    "_bodyText": "",
    "credentials": "include",
    "headers": Headers {
      "map": Object {
        "cvent-event-id": "eventId",
        "cvent-user-type": "standard",
        "httplogpageloadid": "httplogpageloadid",
        "httplogrequestid": "httplogrequestid",
      },
    },
    "method": "GET",
    "mode": null,
    "referrer": null,
    "url": "https://example.com/travelBase/travel-booking/v1/travel-carts/inviteeId?environment=dev",
  },
]
`;

exports[`travelApiClient.getTravelCart matches snapshot 2`] = `
Object {
  "travelCart": Object {
    "bookings": Array [
      Object {
        "attendee": Object {
          "id": "614c34fe-f5f6-49d9-8d6c-5c7bc76ee466",
        },
        "hotelRoomBookings": Array [
          Object {
            "checkinDate": "2015-06-22",
            "checkoutDate": "2015-06-23",
            "hotelId": "b64c34fe-f5f6-49d9-8d6c-5c7bc76ee4b5",
            "id": "00000000-0000-0000-0000-000000000000",
            "preferences": Object {
              "isAdaAccessibilityOpted": false,
              "isRoomSharingOpted": true,
              "isSmokingOpted": true,
              "rewardsCode": "something",
              "roommateDescription": "",
              "specialRequestNotes": "something",
            },
            "requestedAction": "BOOK",
            "roomTypeId": "c26c34fe-f5f6-49d9-8d6c-5c7bc76ee464",
            "travellerInfo": Object {
              "firstName": "",
              "gender": "0",
              "lastName": "",
            },
          },
        ],
        "id": "00000000-0000-0000-0000-000000000000",
        "registrationTypeId": "00000000-0000-0000-0000-000000000000",
      },
    ],
    "eventSnapshotVersion": "def",
    "id": "6b0b6913-b477-4a8c-9e7e-481ce5e4b33a",
    "isForPlanner": false,
    "isForPreview": false,
    "status": "INPROGRESS",
    "travelSnapshotVersion": "abc",
  },
}
`;

exports[`travelApiClient.updateTravelCart matches snapshot 1`] = `
Array [
  Request {
    "_bodyInit": "{\\"id\\":\\"6b0b6913-b477-4a8c-9e7e-481ce5e4b33a\\",\\"status\\":\\"INPROGRESS\\",\\"bookings\\":[{\\"id\\":\\"00000000-0000-0000-0000-000000000000\\",\\"attendee\\":{\\"id\\":\\"614c34fe-f5f6-49d9-8d6c-5c7bc76ee466\\"},\\"registrationTypeId\\":\\"00000000-0000-0000-0000-000000000000\\",\\"hotelRoomBookings\\":[{\\"id\\":\\"00000000-0000-0000-0000-000000000000\\",\\"hotelId\\":\\"b64c34fe-f5f6-49d9-8d6c-5c7bc76ee4b5\\",\\"roomTypeId\\":\\"c26c34fe-f5f6-49d9-8d6c-5c7bc76ee464\\",\\"checkinDate\\":\\"2015-06-22\\",\\"checkoutDate\\":\\"2015-06-23\\",\\"preferences\\":{\\"specialRequestNotes\\":\\"something\\",\\"isSmokingOpted\\":true,\\"rewardsCode\\":\\"something\\",\\"isRoomSharingOpted\\":true,\\"isAdaAccessibilityOpted\\":false,\\"roommateDescription\\":\\"\\"},\\"travellerInfo\\":{\\"gender\\":\\"0\\",\\"firstName\\":\\"\\",\\"lastName\\":\\"\\"},\\"requestedAction\\":\\"BOOK\\"}]}],\\"travelSnapshotVersion\\":\\"abc\\",\\"eventSnapshotVersion\\":\\"def\\",\\"isForPreview\\":false,\\"isForPlanner\\":false}",
    "_bodyText": "{\\"id\\":\\"6b0b6913-b477-4a8c-9e7e-481ce5e4b33a\\",\\"status\\":\\"INPROGRESS\\",\\"bookings\\":[{\\"id\\":\\"00000000-0000-0000-0000-000000000000\\",\\"attendee\\":{\\"id\\":\\"614c34fe-f5f6-49d9-8d6c-5c7bc76ee466\\"},\\"registrationTypeId\\":\\"00000000-0000-0000-0000-000000000000\\",\\"hotelRoomBookings\\":[{\\"id\\":\\"00000000-0000-0000-0000-000000000000\\",\\"hotelId\\":\\"b64c34fe-f5f6-49d9-8d6c-5c7bc76ee4b5\\",\\"roomTypeId\\":\\"c26c34fe-f5f6-49d9-8d6c-5c7bc76ee464\\",\\"checkinDate\\":\\"2015-06-22\\",\\"checkoutDate\\":\\"2015-06-23\\",\\"preferences\\":{\\"specialRequestNotes\\":\\"something\\",\\"isSmokingOpted\\":true,\\"rewardsCode\\":\\"something\\",\\"isRoomSharingOpted\\":true,\\"isAdaAccessibilityOpted\\":false,\\"roommateDescription\\":\\"\\"},\\"travellerInfo\\":{\\"gender\\":\\"0\\",\\"firstName\\":\\"\\",\\"lastName\\":\\"\\"},\\"requestedAction\\":\\"BOOK\\"}]}],\\"travelSnapshotVersion\\":\\"abc\\",\\"eventSnapshotVersion\\":\\"def\\",\\"isForPreview\\":false,\\"isForPlanner\\":false}",
    "credentials": "include",
    "headers": Headers {
      "map": Object {
        "content-type": "application/json",
        "cvent-event-id": "eventId",
        "cvent-user-type": "standard",
        "httplogpageloadid": "httplogpageloadid",
        "httplogrequestid": "httplogrequestid",
      },
    },
    "method": "PUT",
    "mode": null,
    "referrer": null,
    "url": "https://example.com/travelBase/travel-booking/v1/travel-carts/6b0b6913-b477-4a8c-9e7e-481ce5e4b33a?environment=dev",
  },
]
`;

exports[`travelApiClient.updateTravelCart matches snapshot 2`] = `
Object {
  "travelCart": Object {
    "bookings": Array [
      Object {
        "attendee": Object {
          "id": "614c34fe-f5f6-49d9-8d6c-5c7bc76ee466",
        },
        "hotelRoomBookings": Array [
          Object {
            "checkinDate": "2015-06-22",
            "checkoutDate": "2015-06-23",
            "hotelId": "b64c34fe-f5f6-49d9-8d6c-5c7bc76ee4b5",
            "id": "00000000-0000-0000-0000-000000000000",
            "preferences": Object {
              "isAdaAccessibilityOpted": false,
              "isRoomSharingOpted": true,
              "isSmokingOpted": true,
              "rewardsCode": "something",
              "roommateDescription": "",
              "specialRequestNotes": "something",
            },
            "requestedAction": "BOOK",
            "roomTypeId": "c26c34fe-f5f6-49d9-8d6c-5c7bc76ee464",
            "travellerInfo": Object {
              "firstName": "",
              "gender": "0",
              "lastName": "",
            },
          },
        ],
        "id": "00000000-0000-0000-0000-000000000000",
        "registrationTypeId": "00000000-0000-0000-0000-000000000000",
      },
    ],
    "eventSnapshotVersion": "def",
    "id": "6b0b6913-b477-4a8c-9e7e-481ce5e4b33a",
    "isForPlanner": false,
    "isForPreview": false,
    "status": "INPROGRESS",
    "travelSnapshotVersion": "abc",
  },
}
`;
